### PR TITLE
fix(Core/ArenaTeam): Arena team queue protections

### PR DESCRIPTION
**Pull request by aradep**
## CHANGES PROPOSED:
- If an arena team member is removed from the team, all other members are forced to leave the current arena queue.
- Arena team members cannot be removed from the team while invited to an arena match.
- Arena teams cannot be disbanded while a team member is in an arena queue
- Arena teams cannot be disbanded while a team member is invited to an arena match.

I take no credit for this. Just porting @dcombat's [commit](https://github.com/dcombat/azerothcore-wotlk/commit/d034186b4528d9d30c531ac01a525e594da1503a) from [here](https://github.com/azerothcore/azerothcore-wotlk/issues/2720#issuecomment-596447612)

## ISSUES ADDRESSED:
Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2720

## HOW TO TEST THE CHANGES:

Create an arena team and join an arena queue

<details><summary>While in arena queue</summary>

1. Remove a team member

2. Attempt to disband the team

</details>

<details><summary>While invited to arena match</summary>

1. Attempt to remove a team member

2. Attempt to disband the team
</details>

**Edit by Shin:** also try to play with Arenas in general and make sure nothing is broken (queue, teams, scores, etc..).

## Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
good job @aradep ! I think it's good if we test the arenas in general before merging this, just to make sure nothing is broken. I've edited the test instructions.

@azerothcore/testers please have a look at this.

**Comment by pklloveyou** [comment]
Tested

**Comment by FrancescoBorzi** [comment]
thank you @aradep and @pklloveyou

**Comment by FrancescoBorzi** [comment]
@peti446 do you mind reviewing this as you recently worked with arenas?

File: src/server/game/Battlegrounds/ArenaTeam.cpp [hunk 1]
```diff
@@ -308,17 +309,44 @@ void ArenaTeam::SetCaptain(uint64 guid)
...
+        // Remove queues of members
+        if (Player* playerMember = ObjectAccessor::FindPlayerInOrOutOfWorld(itr->Guid))
+        {
+            if (group && playerMember->GetGroup() && group->GetGUID() == playerMember->GetGroup()->GetGUID())
+            {
+                if (BattlegroundQueueTypeId bgQueue = BattlegroundMgr::BGQueueTypeId(BATTLEGROUND_AA, GetType()))
+                {
+                    GroupQueueInfo ginfo;
+                    BattlegroundQueue& queue = sBattlegroundMgr->GetBattlegroundQueue(bgQueue);
+                    if (queue.GetPlayerGroupInfoData(playerMember->GetGUID(), &ginfo))
```
Comment (peti446) [inline src/server/game/Battlegrounds/ArenaTeam.cpp:327]: Add brackets here

File: src/server/game/Handlers/ArenaTeamHandler.cpp [hunk 2]
```diff
@@ -233,6 +234,19 @@ void WorldSession::HandleArenaTeamLeaveOpcode(WorldPacket& recvData)
         return;
     }
 
+    // Player cannot be removed during queues
+    if (BattlegroundQueueTypeId bgQueue = BattlegroundMgr::BGQueueTypeId(BATTLEGROUND_AA, arenaTeam->GetType()))
+    {
+        GroupQueueInfo ginfo;
+        BattlegroundQueue& queue = sBattlegroundMgr->GetBattlegroundQueue(bgQueue);
+        if (queue.GetPlayerGroupInfoData(_player->GetGUID(), &ginfo))
```
Comment (peti446) [inline src/server/game/Handlers/ArenaTeamHandler.cpp:242]: Add brackets

File: src/server/game/Handlers/ArenaTeamHandler.cpp [hunk 3]
```diff
@@ -316,6 +340,19 @@ void WorldSession::HandleArenaTeamRemoveOpcode(WorldPacket& recvData)
         return;
     }
 
+    // Team member cannot be removed during queues
+    if (BattlegroundQueueTypeId bgQueue = BattlegroundMgr::BGQueueTypeId(BATTLEGROUND_AA, arenaTeam->GetType()))
+    {
+        GroupQueueInfo ginfo;
+        BattlegroundQueue& queue = sBattlegroundMgr->GetBattlegroundQueue(bgQueue);
+        if (queue.GetPlayerGroupInfoData(_player->GetGUID(), &ginfo))
```
Comment (peti446) [inline src/server/game/Handlers/ArenaTeamHandler.cpp:348]: Add brackets

**Review by peti446 — APPROVED** [review]
Minor changes rest seems fine and nice addition @aradep 👍

**Review by FrancescoBorzi — APPROVED** [review]
approving on behalf of @peti446

**Comment by pklloveyou** [comment]
Tested

**Comment by Kitzunu** [comment]
Thanks!

**Pull request merged**